src/System.Net.Security/src/System/Net/Security/SslSyncAsyncAdapters.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Security
+{
+    internal interface ISslWriteAdapter
```

Comment:
I wonder if the interface and impls should be nested inside SslStreamInternal.  That's presumably the only place they will ever be used, right?  

If we needed the adapter to be able to call some private method on SslStreamInternal, then it would be nice to have them nested.  As it stands it doesn't matter much, but it might in the future.

Reply:
I am okay with that if it is what you would prefer, however it means eventually 4 structs + 2 interfaces (read/write side async/sync) which is a decent chunk at the bottom of the class. Another option is I could partial class StreamInternal so they are still in another file but have private access?

Reply:
moved it inside SslStreamInternal using a partial class and made the methods private. It also exposed the fact that my SingleChunk and MultipleChunk methods were internal when they could/should be private so I changed that as well.